src/AutoTable.jsx

```diff
@@ -20,73 +34,97 @@ class AutoTable extends React.Component {
+    return null;
+  };
 
-  onSort = (column, direction) => {
-    const { data: { items } } = this.props;
-    const ordering = `${direction === "desc" ? "-" : ""}${column}`;
-    this.setState({ sortColumn: column, sortDirection: direction });
-    items.applyParams({ ordering });
+  getTableBodyForResult = ({ data, error, loading }) => {
+    const { children, columns, maxRows } = this.props;
+
+    if (data) {
+      return children(data);
+    } else if (error) {
+      return <ErrorRow columnCount={columns.length} />;
+    } else if (loading) {
+      return [...Array(maxRows || 1)].map((_, index) => (
+        <LoadingRow key={index} rows={columns.length} /> // eslint-disable-line react/no-array-index-key
+      ));
+    }
+
+    return null;
   };
 
-  onSearch = search => {
-    const { data: { items } } = this.props;
-    items.applyParams({ search });
+  handleSort = (column, direction) => {
+    const ordering = `${direction === "desc" ? "-" : ""}${column}`;
+    this.setState({ ordering, sortColumn: column, sortDirection: direction });
   };
 
+  handleSearch = search => this.setState({ search });
+
+  handleChangePage = page => this.setState({ page });
+
   render() {
     const {
+      action,
       columns,
-      data: { items },
+      filter,
       headerPadding,
       maxRows,
       paginatable,
       searchable,
       sortable
     } = this.props;
-    const { sortColumn, sortDirection } = this.state;
-    const pagination =
-      items.total_count > 0
-        ? {
-            count: items.total_count,
-            rowsPerPage: items.page_size,
-            page: items.page_number,
-            onChangePage: page => items.applyParams({ page })
-          }
-        : null;
+
+    const { ordering, page, search, sortColumn, sortDirection } = this.state;
+
+    const variables = { ...filter, size: maxRows, ordering, page, search };
+
     return (
-      <Table
-        columns={
-          sortable ? columns : columns.map(column => ({ label: column.label }))
-        }
-        headerPadding={headerPadding}
-        onSearch={this.onSearch}
-        onSort={this.onSort}
-        pagination={paginatable ? pagination : null}
-        searchable={searchable}
-        sortColumn={sortColumn}
-        sortDirection={sortDirection}
-      >
-        {items.loading
-          ? [...Array(maxRows || 1)].map((_, index) => (
-              <LoadingRow key={index} rows={columns.length} />
-            ))
-          : this.props.children(items)}
-      </Table>
+      <Data action={action} variables={variables}>
+        {result => (
+          <Table
+            columns={
+              sortable
+                ? columns
+                : columns.map(column => ({ label: column.label }))
+            }
+            headerPadding={headerPadding}
+            onSearch={this.handleSearch}
+            onSort={this.handleSort}
+            pagination={
+              paginatable ? this.getPaginationForData(result.data) : null
```

Comment:
`&&`?

Reply:
I had that originally but then it passes `false` to the underlying component and that caused weirdness. Worth looking into again because that should be something we handle